franklin/builder/models.py

```diff
@@ -208,41 +204,30 @@ class Environment(models.Model):
     def build(self):
         if self.current_deploy:
             is_tag_build = hasattr(self.current_deploy, 'tagbuild')
-            url = os.environ['BUILDER_URL']
+            branch = self.current_deploy.branch if not is_tag_build else ''
+            tag = self.current_deploy.tag if is_tag_build else ''
+            git_hash = self.current_deploy.git_hash if not is_tag_build else ''
+
+            url = os.environ['BUILDER_URL'] + '/build'
             headers = {'content-type': 'application/json'}
             body = {
                 "deploy_key": os.environ['GITHUB_OAUTH'],
-                "branch": self.current_deploy.branch if not is_tag_build else '',
-                "tag": self.current_deploy.tag if is_tag_build else '',
-                "git_hash": self.current_deploy.git_hash if not is_tag_build else '',
+                "branch": branch,
+                "tag": tag,
+                "git_hash": git_hash,
                 "repo_owner": self.site.owner.name,
                 "path": self.current_deploy.path,
                 "repo_name": self.site.name,
                 "environment_id": self.id
                 }
-            r = None
-            try:
-                r = requests.post(url, data=json.dumps(body), headers=headers)
-            except (ConnectionError, HTTPError, Timeout) as e:
-                logger.error('Connection exception : %s', e)
-            except:
-                logger.error('Unexpected Builder error: %s', sys.exc_info()[0])
-
-            if r is not None:
-                if (status.is_success(r.status_code) and
-                        r.headers['Content-Type'] == 'application/json'):
-                    building_status = r.json()['building']
-                    if building_status:
-                        self.status = self.BUILDING
-                        self.save()
-                        return
-                    else:
-                        logger.error("Negative response from Builder")
-                else:
-                    logger.error('Builder responded without json')
+
+            response = make_rest_post_call(url, headers, body)
+            if not status.is_success(response.status_code):
+                logger.error("Negative response from Builder: %s",
+                             response.status_code)
+                self.status = self.FAILED
```

Comment:
This refactor is now compatible with builder. Fixes that bug we saw where API would report a failure even though builder was succeeding.